game/tasks/task04/task_i_the_raft.py

```diff
@@ -4,17 +4,86 @@
 
 > (I) **Interface Segregation Principle**
 >
-> Clients should not be forced to depend upon interfaces that they do not use
+> Clients should not be forced to depend upon interfaces or methods that they do not use
 
 """
+from arcade import Sprite
+from ...core.registration import Register, SpriteRegister
+from ...core.views.vehicle_sprite import VehicleSprite, VehicleType
+from ...core.constants import RAFT_STARTING_X, RAFT_STARTING_Y
 
-"""
 
-This task hasn't been Implemented yet!
+SOURCE_NAME = "task04_I_raft"  # FYI: Required for code reload
+
+"""
+Goal: there are three base classes commented out below! Uncomment the correct one to be used
+with `RaftSprite`. Scroll down to the `RaftSprite` for more details.
 
-FYI: we are planning to make this task the implementation of raft so that the player
-can escape the small island and make it to the mainland. This task could explore how
-making an abstract GameSprite could break the Interface Segregation Principle if
-methods for both the player and vehicle are expected
+Note: You may notice that the base classes in this exercise are slightly different from those in
+the instructions. The base classes here neither inherit from `ABC` (a helper class that has ABCMeta
+as its metaclass) nor use @abstractmethod. This means that the base classes are are not abstract,
+and their methods should not be re-implemented by their subclasses! 
 
+Regardless of whether a base class is abstract of not, its subclasses should not depend on a base
+class that has methods that they do not need.
 """
+
+
+# class VehicleInterface(Sprite):
+#     """Base class for vehicles that move on water."""
+
+#     def __init__(self, sheet_name: str, center_x: int, center_y: int):
+#         super().__init__(sheet_name, center_x=center_x, center_y=center_y)
+
+#     def move_on_water():
+#         ...
+
+#     def move_on_land():
+#         ...
```

Comment:
I think the student task should be to refactor the code to a single `move` method, rather than having both.

At the start, maybe `WaterVehicle.move_on_land` only raises an error and the opposite `LandVehicle.move_on_water` raises an error? When instead, `move` could be just a single method. 

Sort of like this example: https://reflectoring.io/interface-segregation-principle/

Reply:
Actually, my recommendation above is more in-line with LSP

Maybe the task could be to split a single giant base class into two separate classes? Maybe a single base class that implements both water and land motion and something related to dock/disembark/hop-out/climb-in/etc.?

https://drive.google.com/file/d/0BwhCYaYDn8EgOTViYjJhYzMtMzYxMC00MzFjLWJjMzYtOGJiMDc5N2JkYmJi/view?resourcekey=0-3H2Ld4l-dIZZVRmHSpNLcA

Reply:
Yeah I originally wanted the task to be splitting a class, but I also wanted the change to have a visual impact on the game...I continue to think about a way to do both

Reply:
(Also I just noticed - In the description, I wrote the task is based on the instructions, but I meant introductions! Just fixed it)

Reply:
Done! Let me know what you think

Reply:
I'm going to go ahead and merge this since you approved it previously

Reply:
Yeah, I was testing locally and docking was working now